File: selection_sort.py

def selection_sort(arr):
   

   for i in range(len(arr)):
      min_index = i
      for j in range(i,len(arr)):
         
         if arr[min_index] > arr[j]:
            min_index = j
      
      arr[i],arr[min_index] = arr[min_index],arr[i]

a = [25,12,7,10,8,23]
print(a)
selection_sort(a)
print(a)

File: java/SelectionSort.java
import java.util.Arrays;

public class SelectionSort {
    public static void sort(int[] arr)
    {
        int n = arr.length;
        for(int i=0;i<n-1;i++)
        {
            int min = i;
            for(int j=i+1;j<n;j++)
            {
                if(arr[j]<arr[min])
                    min = j;
            }
            if(min!=i)
                swap(arr,min,i);
        }
    }
    private static void swap(int[] a, int j, int i) {
        int temp = a[i];
        a[i] = a[j];
        a[j] = temp;
    }

    public static void main(String[] args) {
        int a[] = {2,5,4,8,9,7,6,1};
        System.out.println(Arrays.toString(a));
        sort(a);
        System.out.println(Arrays.toString(a));
    }
}
